src/components/inject.js

```diff
@@ -152,6 +152,13 @@ Please be sure the component that calls createSource or createToken is within an
       if (options && typeof options === 'object') {
         const {type, ...rest} = options; // eslint-disable-line no-unused-vars
         const specifiedType = typeof type === 'string' ? type : 'auto';
+
+        if (specifiedType === 'auto' && window.console && window.console.warn) {
+          console.warn(
+            'Source type will be required in future versions of react-stripe-elements. Please pass in a valid source type when calling createSource().'
+          );
+        }
+
```

Comment:
Is it worth moving this check lower, to the point where we've actually computed the inferred source type?

Then we could say something like:

```
Inferred source type of 'foo' for createSource(). This behavior will be deprecated in a future version. Please pass the source type to createSource() explicitly.
```